packages/webui/src/components/MonacoEditor.tsx

```diff
@@ -8,22 +9,42 @@ const editorOptions: EditorProps['options'] = {
     enabled: false,
   },
   readOnly: true,
+  scrollBeyondLastLine: false,
   showFoldingControls: 'always',
   lineNumbers: 'off',
 };
 
 export default function MonacoEditor({ value, language }: MonacoEditorProps) {
-  return <Editor value={value} language={language} options={editorOptions} onMount={setEditorTheme} />;
-}
+  const editorRef = useRef<Parameters<OnMount>['0'] | null>(null);
+
+  const executeFolding = () => {
+    if (!editorRef.current) return;
+
+    // fold deeply nested objects
+    editorRef.current?.trigger('fold', 'editor.foldLevel7', {});
+    editorRef.current?.trigger('fold', 'editor.foldLevel6', {});
+    editorRef.current?.trigger('fold', 'editor.foldLevel5', {});
+    editorRef.current?.trigger('fold', 'editor.foldLevel4', {});
+  };
+
+  const onMount: OnMount = (editor, monaco) => {
+    editorRef.current = editor;
+
+    monaco.editor.defineTheme('envy', {
+      base: 'vs',
+      inherit: true,
+      colors: {
+        'editor.background': colors.gray['200'],
+        'editor.lineHighlightBackground': colors.gray['200'],
+      },
+      rules: [],
+    });
+    monaco.editor.setTheme('envy');
+
+    executeFolding();
```

Comment:
When the editor is in view, we get a flash of the folding happening.  Not the end of the world, but is there anything that can be done to avoid it?

![CleanShot 2023-11-13 at 09 20 06](https://github.com/FormidableLabs/envy/assets/17857833/775bb9ea-8af6-4372-993e-be2c063e282c)